### PR TITLE
345 test datatable refactor

### LINKED ISSUE #345: Separate test_datatable.py into multiple files
- the `test_datatable.py` is getting quite long, and we should separate into more files.
- One file could contain the describe testing functions, another with mutual info testing, another with value_counts...

### PR DESCRIPTION
**Pull request by johnbridstrup**
Refactored test_datatable.py into:

- test_datatable.py
- test_datatable_dtype.py
- test_datatable_ltype.py
- test_datatable_semantic_tags.py
- test_datatable_init.py
- test_datatable_indexing.py
- test_datatable_get_set.py
- test_datatable_errors.py

Cleaned up contributing.md, updated to suggest forking prior to implementing fixes and to remind about creating a new branch

Added .idea to .gitignore for pycharm users

- Closes #345

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request.*

### CODE REVIEW

**Comment by johnbridstrup** [comment]
Looking forward to talking with you tomorrow, @gsheni (:

**Review by gsheni — CHANGES_REQUESTED** [review]
Thanks for the contribution @johnbridstrup!

**Comment by gsheni** [comment]
@johnbridstrup Can you also please sign the CLA? Thanks https://cla-assistant.io/alteryx/woodwork?pullRequest=574

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -11,6 +11,7 @@ Release Notes
     * Testing Changes
         * Update branch reference in tests workflow (:pr:`552`)
         * Fixed text on back arrow on install page (:pr:`564`)
+        * Refactor test_datatable.py (:pr:`574`)
 
     Thanks to the following people for contributing to this release:
     :user:`gsheni`, :user:`tamargrey`
```
[the remark this answers is not in the document]
Comment (johnbridstrup) [reply docs/source/release_notes.rst]: Done :)

**Review by gsheni — APPROVED** [review]
Thanks, looks good!

**Pull request merged**